CONTRIBUTING.md

```diff
@@ -207,9 +211,74 @@ To build the documentation locally:
 
 1. Navigate to the `docs` directory.
 2. Run `make html`.
-3. Open "_build/html/index.html" in your browser.
+3. Open "\_build/html/index.html" in your browser to inspect the documentation.
 
 Note that `nbsphinx` currently requires the stand-alone program `pandoc`. If you
-get an error "Pandoc wasn't found", install that pandoc as described in
+get an error "Pandoc wasn't found", install `pandoc` as described in
 [nbsphinx installation
 guide](https://nbsphinx.readthedocs.io/en/0.9.2/installation.html#pandoc).
+
+### Jupyter notebook style guide
+
+If you are contributing changes to the Jupyter notebooks in the documentation, please
+adhere to the following style guidelines.
+
+#### General Jupyter guidelines
+
+Before committing your notebook(s),
+
+- Ensure that you "Restart Kernel and Run All Cells...", making
+  sure that cells are executed in order, the notebook is reproducible and does not have any hidden states.
+- Ensure that the notebook does not raise syntax warnings in the Sphinx build logs as a result of your
+  changes.
+
+#### Links / Cross-references
+
+You are advised to include links in the notebooks as much as possible if it provides the
+reader with more background / context on the topic at hand.
+
+When doing so, never use URL links to refer to other SHAP notebooks, the SHAP
+documentation or other external Sphinx projects. Use [sphinx
+cross-references](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html)
+instead.
+
+We have `sphinx.ext.autosectionlabel` extension enabled, which autogenerates targets
+for every section, named `{relative/path/to/page}:{title-of-section}`.
```

Comment:
It would be helpful here to give an example of a valid cross reference to a notebook.

Reply:
@connortann I couldn't get the cross-reference working using the sphinx targets.

I'm following the `nbsphinx` documentation instead, which recommends [this](https://nbsphinx.readthedocs.io/en/0.9.3/markdown-cells.html#Links-to-Domain-Objects).

It's not as elegant, but it works for now. I recently used it in PR #3336, see the output notebook [here](https://shap.readthedocs.io/en/latest/example_notebooks/tabular_examples/tree_based_models/Force%20Plot%20Colors.html): 

![image](https://github.com/shap/shap/assets/30731072/b0ee02f5-4c5d-4604-998d-d28979cb83c0)

Reply:
I would be good to get usual cross-references working; I'll see if I can take a look later.

Let's not hold up this PR for that though, as this is a good starting point!